vicar/vicarimage.py

```diff
@@ -198,38 +159,54 @@ def __init__(self, source=None, array=None, *, prefix=None, binheader=None):
         self.prefix = prefix
         self.binheader = binheader
 
+    ######################################################################################
+    # Properties
+    ######################################################################################
+
     @property
     def filepath(self):
-        """Get the filepath.
+        """The path to the associated file.
 
         Returns:
-            Path: The path of the VicarImage."""
+            pathlib.Path or None: The path to the associated file, if any.
+        """
+
         return self._filepath
 
     @filepath.setter
     def filepath(self, value):
+        """Set the path to the associated file.
+
+        Parameters:
+            value (pathlib.Path, str, or None):
+                The path to the associated file; None to remove a file association.
+        """
+
         if value is None:
             self._filepath = None
         else:
             self._filepath = pathlib.Path(value)
 
     @property
     def label(self):
-        """Get the VicarLabel object.
+        """The VicarLabel object.
 
         Returns:
-            VicarLabel: The label associated with this VicarImage."""
+            VicarLabel: The VicarLabel object.
+        """
+
         return self._label
 
     @property
     def array2d(self):
-        """Return the array object as 2-D.
+        """The data array as 2-D.
```

Comment:
I don't know why, but none of the properties are showing up with a [Source] link

Reply:
This appears to be a known bug. Here's the issue in the GitHub repo for sphinx.ext.viewcode.
[https://github.com/sphinx-doc/sphinx/issues/11279](https://github.com/sphinx-doc/sphinx/issues/11279)
This is not worth worrying about till they fix it.